lib/jsdom/browser/Window.js

```diff
@@ -216,6 +216,22 @@ function setupWindow(windowInstance, { runScripts }) {
   windowInstance._globalObject = windowInstance;
 }
 
+function makeReplaceablePropertyDescriptor(property) {
+  return {
+    set: Object.getOwnPropertyDescriptor({
+      // eslint-disable-next-line accessor-pairs
+      set [property](value) {
```

Comment:
This ensures that the setter has the correct `name` property.

Reply:
Can you instead define the function and then update its name with `Object.defineProperty`? This is too clever for future readers to understand.